docs/diagrams/wallet-lifecycle/wallet-lifecycle-2-entering-moving-funds-state.tikz

```diff
@@ -0,0 +1,182 @@
+\node[note2, above=of rt_note1.north, anchor=south, yshift=-8mm] (rt_note2)    {Sender incentivized by:\\ \textbf{Fraction of slashed amount}};
+
+% Arrows
+\draw[->]      (rt_start)    -- (rt_check1);
+\draw[->]      (rt_check1)   -| (rt_revert1)  \no;
+\draw[->]      (rt_check1)   -| (rt_check2)   \yes;
+\draw[->]      (rt_check2)   -| (rt_revert1)  \no;
+\draw[->]      (rt_check2)   -| (rt_check3)   \yes;
+\draw[->]      (rt_check3)   -| (rt_revert1)  \no;
+\draw[->]      (rt_check3)   -| (rt_action1)  \yes;
+\draw[->]      (rt_action1)  -- (rt_action2);
+\draw[->]      (rt_action2)  -- (rt_action3);
+\draw[->]      (rt_action3)  -- (rt_action4);
+\draw[->]      (rt_action4)  -- (rt_check4);
+\draw[->, end] (rt_check4)   -| (rt_end1)     \no;
+\draw[->]      (rt_check4)   -| (rt_action5)  \yes;
+\draw[->]      (rt_action5)  -- (rt_check5);
+\draw[->, end] (rt_check5)   -| (rt_end2)     \no;
+\draw[->]      (rt_check5)   -| (rt_check6)   \yes;
+\draw[->]      (rt_check6)   -| (rt_action6)  \no;
+\draw[->]      (rt_action6)  -- (rt_action7);
+\draw[->]      (rt_check6)   -| (rt_action8)  \yes;
+\draw[->]      (rt_action8)  -- (rt_check7);
+\draw[->]      (rt_check7)   -| (rt_action9)  \yes;
+\draw[->]      (rt_check7)   -| (rt_action10) \no;
+\draw[->]      (rt_action9)  |- (rt_action10);
+\draw[->]      (rt_action10) -- (rt_action11);
+
+
+% CLOSEABLE WALLET
+% Nodes
+\node[start, right=of rt_start, xshift=210mm]                    (cw_start)   {Somebody notifies about \textbf{closeable wallet}};
+\node[check, below=of cw_start]                                  (cw_check1)  {Is wallet marked\\ as ACTIVE?};
+\node[check, below left=of cw_check1]                            (cw_check2)  {Is wallet LIVE?};
+\node[revert, below right=of cw_check1]                          (cw_revert1) {Transaction reverted};
+\node[check, below right=of cw_check2]                           (cw_check3)  {Is wallet\\ older than\\ wallet\_max\_age or\\ is wallet's balance below\\ wallet\_min\_closure\_btc?};
+\node[revert, below left=of cw_check3, xshift=-2cm]              (cw_revert2) {Transaction reverted};
+\node[note1, right=of cw_start.south east, anchor=south west]    (cw_note1)   {Can be successfully called for wallets:\\
+                                                                               \textbf{- LIVE}};
+\node[note2, above=of cw_note1.north, anchor=south, yshift=-8mm] (cw_note2)   {Sender incentivized by:\\ \textbf{Reimbursement, if maintainer}};
+
+% Arrows
+\draw[->] (cw_start)  -- (cw_check1);
+\draw[->] (cw_check1) -| (cw_check2)  \no;
+\draw[->] (cw_check1) -| (cw_revert1) \yes;
+\draw[->] (cw_check2) -| (cw_revert2) \no;
+\draw[->] (cw_check2) -| (cw_check3)  \yes;
+\draw[->] (cw_check3) -| (cw_revert2) \no;
+\draw[->] (cw_check3) -| (rt_check6)  \yes;
+
+
+% WALLET HEARTBEAT FAILED
+% Nodes
+\node[start, right=of cw_start, xshift=140mm]                     (whf_start)   {Somebody notifies about \textbf{wallet failing heartbeat}};
```

Comment:
I suggest naming it: 

> Wallet heartbeat failed

Function names is `__ecdsaWalletHeartbeatFailedCallback` but... let's make an exception for this one.